drivers/gpio/gpio_npcx.c

```diff
@@ -18,9 +18,13 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(gpio_npcx, LOG_LEVEL_ERR);
 
-/* GPIO module instances declarations */
-static const struct device *gpio_devs[];
-static int gpio_devs_count;
+/* GPIO module instances */
+#define NPCX_GPIO_DEV(inst) DEVICE_DT_INST_GET(inst),
+static const struct device *const gpio_devs[] = {
+	DT_INST_FOREACH_STATUS_OKAY(NPCX_GPIO_DEV)
+};
+
+static int gpio_devs_count = ARRAY_SIZE(gpio_devs);
```

Comment:
This can be also `const`, but since the value is used only in one place, it would be actually better to use `ARRAY_SIZE(gpio_devs)` directly there.

Reply:
thanks, simplified by dropping gpio_devs_count